trieste/models/gpflow/models.py

```diff
@@ -106,6 +115,19 @@ def __init__(
 
         self._ensure_variable_model_data()
 
+        # Cache posterior for fast sequential predictions.
+        # It is important that we create the posterior *after* we ensure that the model data is
+        # variable.
+        self._posterior = model.posterior(PrecomputeCacheType.VARIABLE)
```

Comment:
So I didnt have `PrecomputeCacheType.VARIABLE`. Is this why it wasn't updating properly?